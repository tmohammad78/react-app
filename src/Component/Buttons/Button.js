import React, { useState } from 'react';
import PropTypes from 'prop-types';
import { ButtonStyle, ButtonAnc } from './style';

import './style.scss';

export const Button = ({ children, ...props }) => {
  return <ButtonStyle {...props}>{children}</ButtonStyle>;
};

Button.defaultProps = {
  primary: true
};

// Button.propTypes = {
//   outline: PropTypes.bool,
//   children: PropTypes.string,
//   color: PropTypes.string,
//   size: PropTypes.string,
//   action: PropTypes.bool,
//   name: PropTypes.string
// };

// export const Button;
// export const AncButton;
