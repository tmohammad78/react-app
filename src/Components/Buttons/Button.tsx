import React from 'react';
import { ButtonStyle } from './style';

export const Button = ({ children, ...props }: any) => {
	return <ButtonStyle {...props}>{children}</ButtonStyle>;
};

Button.defaultProps = {
	primary: true
};
