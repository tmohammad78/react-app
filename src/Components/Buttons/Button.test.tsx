import * as React from 'react';
import { Button } from './Button';
// import ReactDom, { render } from 'react-dom';
import { cleanup, fireEvent, render, screen } from '@testing-library/react';
import { createRenderer } from 'react-dom/test-utils';
import renderer from 'react-test-renderer';

test('render test ', () => {
	const props = { bgcolor: 'red', color: 'white' };
	// ReactDom.render(<Button />, div);
	const { unmount, container, getByText } = render(<Button {...props} >Click</Button>);
	console.log(container);
	expect(getByText('Click')).toBeInTheDocument();
	unmount();
	expect(getByText('Click')).not.toBeInTheDocument();
	// expect(root.props).toEqual({ ...props, primary: true });
	// const tree = component.toJSON();
	// expect(tree).toMatchSnapshot();
	// const div = document.createElement('div');
	// console.log(div.innerHTML);
	// expect(div.querySelector('button')).toHaveAttribute('class');

});
