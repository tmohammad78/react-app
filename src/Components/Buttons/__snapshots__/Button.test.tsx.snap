// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render test  1`] = `
<button
  className="sc-bdfBwQ sc-gsTCUz iKbolA bHKSDE"
  color="white"
/>
`;
